### DISCUSSION TITLE
How does the IntegratedBC work?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#22931] [Q&A General] [answered]
Hi all,

I find that the return value of NeumanBC is negative in NeumannBC.C. Could anyone kindly explain it to me? Why it is negative? 

Best regards,
SomnusYu

**Comment by jiangwen84** [comment]
You can read some basics of finite element method, e.g., weak form, etc. After applying integration by parts and divergence theory, there will be a negative sign for the Neumann boundary conditions.

**Reply by SomnusYu** [reply]
Thank you!

**Comment by friedmud** [comment, marked as answer]
It's due to the fact that we always write a Poisson/Laplacian/Diffusion problem with a negative on the divergence term.  This is a convention because when you integrate by parts to form the weak form you end up with a symmetric positive definite (SPD) matrix.

You can see the whole process here... including the integration by parts that generates the boundary integral (which still has a negative on it).

**Reply by friedmud** [reply]
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step04_weak_form.html#step-4-generate-a-weak-form

**Reply by SomnusYu** [reply]
OK, thank you very much!
